File: Review-04/Example-5/src/main/java/edu/odu/cs/cs330/examples/PlayTicTacToe.java
package edu.odu.cs.cs330.examples;

import java.io.IOException;

/**
 * The wrapper for main function.
 */
public class PlayTicTacToe
{
    /**
     * This program accepts no command line arguments.
     *
     * @param args not used
     */
    public static void main(String[] args)
        throws IOException
    {
        Player tom = new Player("Thomas");
        Player jay = new Player("Jay");

        Game game = new Game(tom, jay);

        //while(!(game.getBoard().isFull())) {
        while (game.isNotOver()) {
            game.playRound();
        }

        System.out.println(game.getBoard());

        if (game.endedWithWin()) {
            System.out.printf("Congratulations %s!%n", game.getWinner());
        }
    }
}

File: Review-04-Python-Unit-Testing/Example-1/play_tictactoe.py
#! /usr/bin/env python3

"""
This is a demo driver for a basic single round game of Tic-Tac-Toe.
"""

from examples import Game, Player


def main():
    """
    This is a simple main function that takes no command line arguments.
    """

    tom = Player("Thomas")
    jay = Player("Jay")

    game = Game(tom, jay)

    while game.is_not_over():
        game.play_round()

    print(game.get_board())

    if game.ended_with_win():
        print("Congratulations {}!".format(game.get_winner()))


if __name__ == "__main__":
    main()

    # @todo add exception handling
